apps/examples/performance/tls_seclink/tls_seclink.c

```diff
@@ -0,0 +1,165 @@
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <pthread.h>
+#include "tls_common.h"
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int tls_seclink_main(int argc, char *argv[])
+#endif
+{
+	int ret;
+	int srv_ret = -1;
+	int cli_ret = -1;
+	struct timespec ts_origin;
+	struct timespec ts;
+
+	clock_gettime(CLOCK_REALTIME, &ts_origin);
+
+	ts.tv_sec = 1612483200; // 2021-02-05 00:00:00
+	ts.tv_nsec = 0;
+
+	clock_settime(CLOCK_REALTIME, &ts);
+
+	printf("-- [TLS SECLINK TEST] START --\n");
+
+	pthread_t tls_server;
+	pthread_attr_t attr;
+	struct sched_param sparam;
+	int status;
+
+	/* Initialize the attribute variable */
+	status = pthread_attr_init(&attr);
+	if (status != 0) {
+		TLS_SECLINK_LOG("tls_seclink(): pthread_attr_init failed, status=%d\n", status);
+		goto exit;
+	}
+
+	/* 1. set a priority */
+	sparam.sched_priority = TLS_SERVER_PRI;
+	status = pthread_attr_setschedparam(&attr, &sparam);
+	if (status != 0) {
+		TLS_SECLINK_LOG("tls_seclink(): pthread_attr_setschedparam failed, status=%d\n", status);
+		goto exit;
+	}
+
+	/* 2. set a stacksize */
+	status = pthread_attr_setstacksize(&attr, TLS_STACKSIZE);
+	if (status != 0) {
+		TLS_SECLINK_LOG("tls_seclink(): pthread_attr_setstacksize failed, status=%d\n", status);
+		goto exit;
+	}
+
+	/* 3. set a sched policy */
+	status = pthread_attr_setschedpolicy(&attr, SCHED_RR);
+	if (status != 0) {
+		TLS_SECLINK_LOG("tls_seclink(): pthread_attr_setschedpolicy failed, status=%d\n", status);
+		goto exit;
+	}
+
+	/* 4. create pthread with entry function */
+	status = pthread_create(&tls_server, &attr, tls_server_seclink_main, NULL);
+	if (status != 0) {
+		TLS_SECLINK_LOG("tls_seclink(): pthread_create failed, status=%d\n", status);
+		goto exit;
+	}
+
+	status = pthread_setname_np(tls_server, "tls_seclink_server");
+	if (status != 0) {
+		TLS_SECLINK_LOG("failed to set name for pthread(%d)\n", status);
+		goto exit;
+	}
+
+	sleep(1);
+
+	cli_ret = (int)tls_client_seclink_main(NULL);
+	TLS_SECLINK_LOG("[CLIENT] End : %d --\n", cli_ret);
+
+	if (cli_ret != 0) {
+		int sockfd = socket(AF_INET, SOCK_STREAM, 0);
+		struct sockaddr_in server;
+		server.sin_family = AF_INET;
+		server.sin_port = atoi(SERVER_PORT);
+		server.sin_addr.s_addr = inet_addr(SERVER_ADDR);
+
+		cli_ret = connect(sockfd, (struct sockaddr *)&server, sizeof(struct sockaddr_in));
```

Comment:
How about to comment why did you call connect here.

Reply:
@pillip8282 
If cli_ret is not 0, the server thread might keep waiting connection from a client.
Hence, to be clear that the server thread is stopped, making temporary connection with the server.

I also remained the comment on the code.